src/main/java/it/gov/pagopa/admissibility/rest/anpr/residence/ResidenceAssessmentRestClientImpl.java

```diff
@@ -0,0 +1,135 @@
+import it.gov.pagopa.admissibility.rest.anpr.AgidJwtSignature;
+import it.gov.pagopa.admissibility.rest.anpr.AnprWebClient;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.MediaType;
+import org.springframework.http.client.reactive.ReactorClientHttpConnector;
+import org.springframework.stereotype.Service;
+import org.springframework.web.reactive.function.client.WebClient;
+import reactor.core.publisher.Mono;
+import reactor.netty.http.client.HttpClient;
+
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.text.SimpleDateFormat;
+import java.util.Base64;
+import java.util.Date;
+
+@Service
+@Slf4j
+public class ResidenceAssessmentRestClientImpl implements ResidenceAssessmentRestClient{
+    private final WebClient webClient;
+    private final AgidJwtSignature agidJwtSignature;
+
+    private final String headRequestSenderCode;
+    private final String headRequestAddresseeCode;
+    private final String headRequestOperationRequest;
+    private final String headRequestSendType;
+
+    private final String dataRequestPersonalDetailsRequest;
+
+    private final ObjectMapper objectMapper;
+
+    @SuppressWarnings("squid:S00107") // suppressing too many parameters constructor alert
+    public ResidenceAssessmentRestClientImpl(AnprWebClient anprWebClient,
+                                             AgidJwtSignature agidJwtSignature,
+
+                                             @Value("${app.anpr.c020-residenceAssessment.base-url}") String residenceAssessmentBaseUrl,
+
+                                             @Value("${app.anpr.c020-residenceAssessment.properties.headRequest.senderCode}") String headRequestSenderCode,
+                                             @Value("${app.anpr.c020-residenceAssessment.properties.headRequest.addresseeCode}") String headRequestAddresseeCode,
+                                             @Value("${app.anpr.c020-residenceAssessment.properties.headRequest.operationRequest}") String headRequestOperationRequest,
+                                             @Value("${app.anpr.c020-residenceAssessment.properties.headRequest.sendType}") String headRequestSendType,
+
+                                             @Value("${app.anpr.c020-residenceAssessment.properties.dataRequest.personalDetailsRequest}") String dataRequestPersonalDetailsRequest,
+
+                                             ObjectMapper objectMapper) {
+        this.agidJwtSignature = agidJwtSignature;
+
+        this.objectMapper = objectMapper;
+
+        HttpClient httpClient = anprWebClient.getHttpClientSecure();
+
+        this.webClient = WebClient.builder()
+                .clientConnector(new ReactorClientHttpConnector(httpClient))
+                .baseUrl(residenceAssessmentBaseUrl)
+                .build();
```

Comment:
The AnprWebClient will provide the builder to which just clone and define baseUrl